packages/common-all/src/engine/EngineV3Base.ts

```diff
@@ -43,41 +48,75 @@ export abstract class EngineV3Base implements ReducedDEngine {
   /**
    * See {@link DEngine.getNote}
    */
-  async getNote(id: string): Promise<RespV3<NoteProps>> {
+  async getNote(id: string): Promise<GetNoteResp> {
     return this.noteStore.get(id);
   }
 
+  /**
+   * See {@link DEngine.bulkGetNotes}
+   */
+  async bulkGetNotes(ids: string[]): Promise<BulkGetNoteResp> {
+    const bulkResponses = await this.noteStore.bulkGet(ids);
+
+    const errors = bulkResponses
+      .flatMap((response) => response.error)
+      .filter(isNotUndefined);
+
+    return {
+      error: errors.length > 0 ? new DendronCompositeError(errors) : undefined,
+      data: bulkResponses
+        .flatMap((response) => response.data)
+        .filter(isNotUndefined),
+    };
+  }
+
+  /**
+   * See {@link DEngine.bulkGetNotesMeta}
+   */
+  async bulkGetNotesMeta(ids: string[]): Promise<BulkGetNoteMetaResp> {
+    const bulkResponses = await this.noteStore.bulkGetMetadata(ids);
+
+    const errors = bulkResponses
+      .flatMap((response) => response.error)
+      .filter(isNotUndefined);
+
+    return {
+      error: errors.length > 0 ? new DendronCompositeError(errors) : undefined,
+      data: bulkResponses
+        .flatMap((response) => response.data)
+        .filter(isNotUndefined),
+    };
+  }
+
   /**
    * See {@link DEngine.findNotes}
    */
-  async findNotes(opts: FindNoteOpts): Promise<NoteProps[]> {
+  async findNotes(opts: FindNoteOpts): Promise<FindNotesResp> {
     const resp = await this.noteStore.find(opts);
     return resp.data ? resp.data : [];
   }
 
   /**
    * See {@link DEngine.findNotesMeta}
    */
-  async findNotesMeta(opts: FindNoteOpts): Promise<NotePropsMeta[]> {
+  async findNotesMeta(opts: FindNoteOpts): Promise<FindNotesMetaResp> {
     const resp = await this.noteStore.findMetaData(opts);
     return resp.data ? resp.data : [];
   }
 
   /**
    * See {@link DEngine.bulkWriteNotes}
    */
-  async bulkWriteNotes(
-    opts: BulkWriteNotesOpts
-  ): Promise<Required<BulkResp<NoteChangeEntry[]>>> {
+  async bulkWriteNotes(opts: BulkWriteNotesOpts): Promise<BulkWriteNoteResp> {
```

Comment:
super nit: for consistency, can we change the resp type to `BulkWritNote*s*Resp`